cxx-qt-lib/build.rs

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use quote::ToTokens;
+use std::io::Write;
+
+fn main() {
+    let opt = cxx_gen::Opt::default();
+
+    // Read the cargo folder and out folder
+    let dir_manifest = std::env::var("CARGO_MANIFEST_DIR").expect("Could not get manifest dir");
```

Comment:
```suggestion
    let mut dir_manifest = env::var("CARGO_MANIFEST_DIR").expect("Could not get manifest dir");
    if cfg!(windows) {
        dir_manifest = dir_manifest.replace('\\', "/");
    }
```
The `format!` macros use `/` separators but `CARGO_MANIFEST_DIR` uses `\` on Windows.

Reply:
Ugh :-) @Be-ing Maybe split out your commit in #107 that has the manifest_dir() method into a separate pull request. Then we can land that first and rebase other things on top as they'll take longer to review :-)

Reply:
#108

Reply:
@Be-ing if the CARGO_MANIFEST_DIR is never used by CMake and only used by Rust's `std::fs::File::Create(path)` etc, do those need to have the \ to / flip as well ?

Reply:
Good question. I don't know. I wouldn't assume that the Rust standard library's Path works with mixed \ and / separators though.

Reply:
Done for now.